src/i_sound.c

```diff
@@ -717,22 +735,30 @@ boolean IsMus(byte *mem, int len)
 
 void *I_RegisterSong(void *data, int size)
 {
+    int i;
+
     if (IsMus(data, size) || IsMid(data, size))
     {
-        if (midi_player_module)
-        {
-            active_module = midi_player_module;
-            return active_module->I_RegisterSong(data, size);
-        }
+        active_module = midi_player_module;
+        return active_module->I_RegisterSong(data, size);
     }
 
     if (midi_player_module == &music_opl_module)
```

Comment:
I always choke on this. Could you add a comment that this is the non-MIDI code path? It's easy to miss the `return` statement in the condition above.

Reply:
Done in https://github.com/fabiangreffrath/woof/pull/976/commits/fa635ac4161ef89d2528b251bada8eebd3c3d9c9